flake.nix

```
@@ -0,0 +1,58 @@
+{
+  description = "Codex build flake";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+
+        # Create a dummy lsb_release script
+        dummy-lsb-release = pkgs.writeScriptBin "lsb_release" ''
+          #!${pkgs.stdenv.shell}
+          echo "Distributor ID: NixOS"
+          echo "Description:    NixOS"
+          echo "Release:        Unstable"
+          echo "Codename:       nixos"
+        '';
+      in {
+        devShell = pkgs.mkShell {
+            buildInputs = with pkgs; [
+                gcc11
+                cmake
+                git
+                rustc
+                cargo
+                bash
+                pkg-config
+                openssl
+                gmp
+                dummy-lsb-release
+                nim
+                nimble
+                nodejs_18
+                llvmPackages.openmp
+            ];
```

I don't think separating it all like that into multiple lines is great. I'd do something more organized:
```suggestion
            buildInputs = with pkgs; [
              # General
              git pkg-config openssl lsb-release
              # Build
              rustc cargo nim nimble gcc11 cmake
              # Libraries
              gmp llvmPackages.openmp
              # What is this for?
              nodejs_18
            ];
```

---

The nodejs18 is used for running tests https://docs.codex.storage/learn/local-two-client-test . I assumed if it is on their website listed as official test that it would be neat to have that possibility in the shell.